nc2zarr/opener.py

```diff
@@ -91,9 +94,15 @@ def _open_mfdataset(
                 combine = 'by_coords'
                 warnings.warn(f'input/concat_dim is not specified, '
                               f'combining by coordinates')
+            if 's3' in input_paths[0]:
+                fileset = [s3.open(input_path) for input_path in input_paths]
+                engine = 'h5netcdf'
```

Comment:
The assumption of `engine = 'h5netcdf'` if the filesystem is AWS S3 is a very special case (your case) that must be generalized.

The engine used should be a setting in the configuration file.